packages/react-dom/src/server/ReactDOMFizzServerBrowser.js

```diff
@@ -16,14 +16,19 @@ import {
   abort,
 } from 'react-server/src/ReactFizzServer';
 
-type Controls = {
-  stream: ReadableStream,
-  abort(): void,
+type Options = {
+  signal?: AbortSignal,
 };
 
-function renderToReadableStream(children: ReactNodeList): Controls {
+function renderToReadableStream(
+  children: ReactNodeList,
+  options?: Options,
+): ReadableStream {
   let request;
-  const stream = new ReadableStream({
+  if (options && options.signal) {
+    options.signal.addEventListener('abort', () => abort(request));
```

Comment:
This doesn't ever need to be cleaned up, right?

Reply:
Good question. I would assume the owner would drop the signal when they're done but we might as well clean it up to be sure.